### DISCUSSION TITLE
Check Gmail Exist but contain characters as ".","+"

### DISCUSSION THREAD

**Discussion by qtzznn1** [laravel/framework#38957] [Q&A] [answered]
I used rule exists for check Email has in database or not
https://laravel.com/docs/8.x/validation#rule-exists

But I realize in Gmail can add characters as ".","+" without affecting the original Email
It means,  a User can create so many account without use other Email
My question is
- How do I can validate email exist in database which can prevent the above case?

**Comment by dennisprudlo** [comment, marked as answer]
You can create your own `gmail_exists` rule with whatever unique ruleset gmail uses

**Reply by qtzznn1** [reply]
I know create new a rule, but in this issue, I want you to suggest a solution for check exist email contains characters ".","+"
Please help me!

**Reply by dennisprudlo** [reply]
If I understood correctly dots and plus symbols are irrelevant in an email? So `john.doe` = `john+doe` = `jo.hn+doe`?

If so just get all emails, remove the symbols and check if the given email already exists.

```php
$input = 'john.doe@example.com';

$input = preg_replace('[\+\.]', '', $input);
$existingEmails = User::pluck('email')->map(fn ($email) => preg_replace('[\+\.]', '', $email));

$existingEmails->contains($input);
```

Somehow like that (I can not guarantee this code will work I typed it on my phone so maybe I forgot something but you should get the idea).

**Comment by tpetry** [comment]
You should create a migration and create an `email_normalized` column where you normalize the email. The benefit is you can do multiple normalizations as there are more email providers with features like this. The `+` behavior is available for many email providers.
